crm-plugin/controllers/activity.js

```diff
@@ -0,0 +1,143 @@
+"use strict";
+
+/**
+ * crm-plugin.js controller
+ *
+ * @description: A set of functions called "actions" of the `crm-plugin` plugin.
+ */
+const { sanitizeEntity } = require("strapi-utils");
+module.exports = {
+  /**
+   * Default action.
+   *
+   * @return {Object}
+   */
+
+  index: async (ctx) => {
+    // Add your own logic here.
+    // Send 200 `ok`
+    ctx.send({
+      message: "ok",
+    });
+  },
+
+  find: async (ctx) => {
+    let activity;
+    try {
+      if (ctx.query._q) {
+        activity = await strapi
+          .query("activity", "crm-plugin")
+          .search(ctx.query);
+      } else {
+        activity = await strapi.query("activity", "crm-plugin").find(ctx.query);
+      }
+      return activity.map((entity) =>
+        sanitizeEntity(entity, {
+          model: strapi.plugins["crm-plugin"].models["activity"],
+        })
+      );
+    } catch (error) {
+      console.error(error);
+      return ctx.badRequest(null, error.message);
+    }
+  },
+
+  findOne: async (ctx) => {
+    const findOneParams = ["id"];
+    const result = strapi.plugins["crm-plugin"].services.utils.checkParams(
+      ctx.params,
+      findOneParams
+    );
+    try {
+      if (!result.error) {
+        const { id } = ctx.params;
+        const entity = await strapi
+          .query("activity", "crm-plugin")
+          .findOne({ id });
+        return sanitizeEntity(entity, {
+          model: strapi.plugins["crm-plugin"].models["activity"],
+        });
+      } else {
+        if (result.error) {
+          return ctx.badRequest(null, result.message);
+        }
+      }
+    } catch (error) {
+      console.error(error);
+      return ctx.badRequest(null, error.message);
+    }
+  },
+
+  create: async (ctx) => {
+    let activityassignee;
+    let activity;
+
+    try {
+      if (ctx.params.id) {
+        if (ctx.request.body.contact) {
+          let activityDetail = {
+            activity: ctx.params.id,
+            contact: ctx.request.body.contact,
+          };
+          activityassignee = await strapi
+            .query("activityassignee", "crm-plugin")
```

Comment:
issue here. what if activity assignee is passed to update api first time. this will fail.